circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/DropTableService.java

```diff
@@ -25,28 +28,69 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.hotels.bdp.circustrain.api.data.DataManipulationClient;
 import com.hotels.hcommon.hive.metastore.client.api.CloseableMetaStoreClient;
+import com.hotels.hcommon.hive.metastore.iterator.PartitionIterator;
 
 public class DropTableService {
 
   private static final Logger LOG = LoggerFactory.getLogger(DropTableService.class);
   private static final String EXTERNAL_KEY = "EXTERNAL";
   private static final String IS_EXTERNAL = "TRUE";
+  private static final short BATCH_SIZE = (short) 10;
 
   /**
    * Removes all parameters from a table before dropping the table.
    */
-  public void removeTableParamsAndDrop(
+  public void removeTableParamsAndDrop(CloseableMetaStoreClient client, String databaseName, String tableName)
```

Comment:
How is this different to `dropTable` now? If we're always going to remove the params in that method then I'm not sure we need this one anymore?

Reply:
Its technically no different, but it gets the table object and passes that into the `dropTable` method. I added the `dropTable` method and made it take in a table object because otherwise when `dropTableAndData` is called it would get the table object, delete partition info, then call `dropTable` which will get the table object again and then drop it. So by doing it this way it reduces the number of times that `client.getTable` is called.